app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/variations/variations.js

```diff
@@ -406,17 +406,20 @@ define([
          *   - configurable-matrix-serialized;
          *   - associated_product_ids_serialized.
          */
-        serializeData: function () {
-            this.source.data['configurable-matrix-serialized'] =
-                JSON.stringify(this.source.data['configurable-matrix']);
-
-            delete this.source.data['configurable-matrix'];
-
-            this.source.data['associated_product_ids_serialized'] =
-                JSON.stringify(this.source.data['associated_product_ids']);
-
-            delete this.source.data['associated_product_ids'];
-        },
+         serializeData: function () {
+             this.source.data['configurable-matrix-serialized'] =
+                 JSON.stringify(this.source.data['configurable-matrix']);
+             if ($('.admin__field-error').length===0) {
```

Comment:
Hi, @ShubhamSharma967. Thanks for the collaboration. We try to not use selectors directly. You can pass the selector name as configuration property to this component.  
Feel free to ask questions or request additional information or some help.